### PR TITLE
added single video page

### PR DESCRIPTION
**Pull request by aishwarya-bm**
1.single video page to view video
2.video details added
3.notes section-not functional yet

https://user-images.githubusercontent.com/66695068/161381086-b6db4f10-ccb0-4491-a80b-af50c4622afc.mp4

### CODE REVIEW

File: src/pages/videolist/VideoPage.jsx [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+  const [video, setVideo] = useState({});
+  const { id } = useParams();
+
+  async function getVideoDetails() {
+    try {
+      const { data } = await axios.get(`/api/video/${id}`);
+      setVideo(() => data.video);
+    } catch (e) {
+      console.log("error", e);
+    }
```
Comment (raman1999) [inline src/pages/videolist/VideoPage.jsx:21]: remove this console

File: src/App.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import { Route,Routes } from "react-router-dom";
 import "./App.css";
-import {Home,History,WatchLater,Liked,ErrorPage,Videolist, LoginPage } from "./pages/index";
+import {Home,History,WatchLater,Liked,ErrorPage,Videolist, LoginPage, VideoPage } from "./pages/index";
```
Comment (raman1999) [inline src/App.js:4]: no need of  pages/index, directly can do pages as it  by default imports from index

**Review by raman1999 — APPROVED** [review]
Ur UI looks great. I liked the way u created separate function for each functionality, it looks more readable,  good naming conventions used, code looks clean

File: src/pages/videolist/VideoPage.jsx (hunk 1 above)
Comment (ttahm3d) [reply src/pages/videolist/VideoPage.jsx:21]: I think that's for error handling

File: src/pages/videolist/VideoPage.jsx [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  }
+  useEffect(() => getVideoDetails(), []);
+  return (
+    <>
+      <Header />
+      <SideNav />
```
Comment (ttahm3d) [inline src/pages/videolist/VideoPage.jsx:27]: This is not react way of doing things. You define the common components once in the file where you are defining routes

```JSX
<BrowserRouter>
     <Header />
     <SIdebar />
     <Routes>
           <Route  />
           <Route />
     <Routes />
</BrowserRouter>
```

This way the Header and sidebar are always rendered and the route matching with the address in address bar will be displayed along with them

**Review by ttahm3d — COMMENTED** [review]
Nicely done.

Have added a comment please do take a look.

**Review by samyakshah3008 — APPROVED** [review]
Good use of shorthand property 
Good folder structure 
Clean UI
 
 Good for approval ✅

File: src/App.js (hunk 2 above)
Comment (aishwarya-bm) [reply src/App.js:4]: Hey Raman, in index im _exporting_ all page components so I gave pages/index

File: src/pages/videolist/VideoPage.jsx (hunk 3 above)
Comment (aishwarya-bm) [reply src/pages/videolist/VideoPage.jsx:27]: Oh i did not know this, also I might not need these in some components so in that case what is the better way?

**Pull request merged**